### ISSUE TITLE
Reloading model causes issue

### ISSUE THREAD

**Issue by AlexanderMath** [rosinality/sagan-pytorch#8]
I save model during training with 

G: 1.42529; D: 1.17999

After loading discriminator and generator I get 

G: -26.45726; D: 0.89195

It looks like discriminator behaves differently. Removing 'd_optimizer.step' gives 

G: 1.10145; D: 1.05196

So it looks like updating the discriminator after re-loading model alters discriminator training. Any idea what could be the cause of this? 

![image](https://user-images.githubusercontent.com/8614529/104312836-74d8d000-54d7-11eb-937f-fe4b7a0d62da.png)

**Comment by AlexanderMath** [comment]
Hmm, it might be that the ```@static``` part of SpectralNorm breaks somehow.

**Comment by AlexanderMath** [comment]
I tried with tree different checkpoints at 10k and 20k iterations, the issue persists. The only difference I found is that sometimes the generator loss becomes very large (+60) instead of very low (-26). All generators produce "nice" images, but after a few steps they completely break down.

**Comment by rosinality** [comment]
Yes, maybe it is spectral norm related error. It would be better to use `torch.nn.utils.spectral_norm`.

**Comment by AlexanderMath** [comment]
Do you think it would be sufficient to remove SpectralNorm class and add a line ```SpectralNorm = torch.nn.utils.spectral_norm```?

**Comment by rosinality** [comment]
Or you can replace https://github.com/rosinality/sagan-pytorch/blob/master/model_resnet.py#L64 this with `torch.nn.utils.spectral_norm`.
